### Lets compute using the Gaussian Equation

In [31]:
from math import *
O = int(input("Input your variance: "))
x = int(input("Input your x value: "))
y = int(input("Input your mean value: "))
f = (1/sqrt(2*pi*O)) * exp((-1/2)*((x - y)**2/O))

Input your variance: 4
Input your x value: 8
Input your mean value: 10


In [32]:
print(f)

0.12098536225957168


#### Lets use functions

In [34]:
def f(mu, sigma2, x): 
    return 1/sqrt(2*pi*sigma2) * exp(-0.5*((x - mu)**2/sigma2))

In [35]:
print(f(10, 4, 8))

0.12098536225957168


But how do i modify **x** to return the maximum value of **f**

To maximize the gaussian dist, **x** will have the same value as **mu**

In [36]:
print(f(10, 4, 10))

0.19947114020071635


Now let write a code to calculate the new mean and variance

In [46]:
def update(mu, r2, sigma2, m_mu):
    new_mean = 1/(sigma2 + r2)*((r2*mu) + (sigma2*m_mu))
    new_variance = 1/(1/sigma2 + 1/r2)
    return [new_mean, new_variance]

In [47]:
print(update(10, 4, 4, 12))

[11.0, 2.0]


notice that the variance terms is half the size of the previous variance terms

Assuming we have new values

In [48]:
print(update(10, 2, 8, 13))

[12.4, 1.6]


Notice that 12.4 is more closer to 13 and we have a slimmer variance, meaning we have a higher peak